src/backend/distributed/planner/relation_restriction_equivalence.c

```diff
@@ -1832,13 +1838,128 @@ FilterPlannerRestrictionForQuery(PlannerRestrictionContext *plannerRestrictionCo
 }
 
 
+/*
+ * GetRestrictInfoListForRelation gets a range table entry and planner
+ * restriction context. The function returns a list of expressions that
+ * appear in the restriction context for only the given relation. And,
+ * all the varnos are set to 1.
+ */
+List *
+GetRestrictInfoListForRelation(RangeTblEntry *rangeTblEntry,
+							   PlannerRestrictionContext *plannerRestrictionContext)
+{
+	RelationRestriction *relationRestriction =
+		RelationRestrictionForRelation(rangeTblEntry, plannerRestrictionContext);
+	if (relationRestriction == NULL)
+	{
+		return NIL;
+	}
+
+	RelOptInfo *relOptInfo = relationRestriction->relOptInfo;
+	List *joinRestrictInfo = relOptInfo->joininfo;
+	List *baseRestrictInfo = relOptInfo->baserestrictinfo;
+
+	List *joinRestrictClauseList = get_all_actual_clauses(joinRestrictInfo);
+	if (ContainsFalseClause(joinRestrictClauseList))
+	{
+		/* found WHERE false, no need  to continue */
+		return copyObject((List *) joinRestrictClauseList);
+	}
+
+
+	List *restrictExprList = NIL;
+	RestrictInfo *restrictInfo = NULL;
+	foreach_ptr(restrictInfo, baseRestrictInfo)
+	{
+		Expr *restrictionClause = restrictInfo->clause;
+
+		/* we cannot process Params beacuse they are not known at this point */
+		if (FindNodeMatchingCheckFunction((Node *) restrictionClause, IsParam))
+		{
+			continue;
+		}
+
+		/*
+		 * If the restriction involves multiple tables, we cannot add it to
+		 * input relation's expression list.
+		 */
+		Relids varnos = pull_varnos((Node *) restrictionClause);
+		if (bms_num_members(varnos) != 1)
+		{
+			continue;
+		}
+
+		/*
+		 * We're going to add this restriction expression to a subquery
+		 * which consists of only one relation in its jointree. Thus,
+		 * simply set the varnos accordingly.
+		 */
+		Expr *copyOfRestrictClause = (Expr *) copyObject((Node *) restrictionClause);
+		List *varClauses = pull_var_clause_default((Node *) copyOfRestrictClause);
+		Var *column = NULL;
+		foreach_ptr(column, varClauses)
```

Comment:
We already have `UpdateVarNosInNode ` call, do we strictly need these assignments? It feels this function does it unnecessarily

Reply:
I guess this is necessary because the other one updates the var nos in the quals